website/docs/developer/research/research-orca-delete-functionality.md

```diff
@@ -0,0 +1,122 @@
+---
+id: research-orca-delete-functionality
+title: Research Notes on ORCA delete functionality.
+description: Research Notes on implementing delete functionality for ORCA.
+---
+
+## Delete functionality use cases
+
+Some use cases for delete functionality are as follows:
```

Comment:
Another thing to think about is can we undo a delete if it is a soft delete?

Reply:
if the bucket has versioning enabled, then yes we can restore the object. The object will be marked for deletion but will not be actually deleted from the bucket. 

If versioning is not enabled, then we cannot restore the object.

Reply:
ok, but in a soft delete scenario we have only changed a record in the database, no buckets have been touched at that point correct?

Reply:
yes it will mark as pending deletion in the DB